ArmEmulator/Mem.fs

```diff
@@ -803,42 +809,55 @@ module Mem
-    let EQUexec (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: LabelInstr) = 
-        abstractResults (Ok (Ok dP, updateSymbolTable symbolTab inputRecord inputRecord.EQUExpr))
+    let execEQU (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: LabelInstr) = 
+        inputRecord.EQUExpr
+        |> updateSymbolTable symbolTab inputRecord
+        |> fun b -> (Ok dP, b)
+        |> Ok
+        |> abstractResults 
     
     ///Executes a Fill instruction, updates Symbol Table and DataPath
-    let FILLexec (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: LabelInstr) = 
-        abstractResults (Ok ((updateMemoryDataPath inputRecord dP), updateSymbolTable symbolTab inputRecord ((inputRecord.FillN))))
+    let execFILL (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: LabelInstr) = 
+        inputRecord.FillN
+        |> updateSymbolTable symbolTab inputRecord 
+        |> fun b -> (updateMemoryDataPath inputRecord dP, b)
+        |> Ok
+        |> abstractResults 
 
 
 
 
-    let ADRexec (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: ADRInstr) = 
+    let execADR (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: ADRInstr) = 
         let updateRegMap (dP: DataPath<'INS>) (inputRecord: ADRInstr) = 
             resultDotBindTwoInp (updateRegister dP) inputRecord.DestReg inputRecord.SecondOp 
-        abstractResults (Ok (Result.map (updateDataPathRegs dP) (updateRegMap dP inputRecord), Ok symbolTab))
-    
+        inputRecord
+        |> updateRegMap dP 
+        |> Result.map (updateDataPathRegs dP) 
+        |> fun a -> (a, Ok symbolTab)
+        |> Ok
+        |> abstractResults 
+
         
 
     let execInstr (dP: DataPath<'INS>) (symbolTab: SymbolTable) (parseInputRecord: Parse<ReturnInstr>) =
         let inputRecord = parseInputRecord.PInstr
         let labelInstructionsHandler (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: LabelInstr) =
             let matchLabIns x =
                 match x with
-                | EQU -> (EQUexec symbolTab dP inputRecord)
-                | DCD -> (DCDexec symbolTab dP inputRecord)
-                | FILL -> (FILLexec symbolTab dP inputRecord)
+                | EQU -> (execEQU symbolTab dP inputRecord)
+                | DCD -> (execDCD symbolTab dP inputRecord)
+                | FILL -> (execFILL symbolTab dP inputRecord)
             Result.bind matchLabIns inputRecord.InstructionType
         let memInstructionsHandler (symbolTab: SymbolTable) (dP: DataPath<'INS>) (inputRecord: MemInstr) =
             let matchMemIns x =
                 match x with
-                | LDR -> (LDRexec symbolTab dP inputRecord)
-                | STR -> (STRexec symbolTab dP inputRecord)
+                | LDR -> (execLDR symbolTab dP inputRecord)
+                | STR -> (execSTR symbolTab dP inputRecord)
             Result.bind matchMemIns inputRecord.InstructionType
 
         match inputRecord with
```

Comment:
I know that `abstractResults` puts things in the correct format, but does it return a `SymbolTable` or a `Map<string, uint32>`? It needs to be a `SymbolTable`